automation/src/test/scala/org/broadinstitute/dsde/test/api/BillingApiSpec.scala

```diff
@@ -0,0 +1,129 @@
+package org.broadinstitute.dsde.test.api
+
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import org.broadinstitute.dsde.workbench.auth.{AuthToken, AuthTokenScopes}
+import org.broadinstitute.dsde.workbench.config.{Credentials, ServiceTestConfig, UserPool}
+import org.broadinstitute.dsde.workbench.fixture._
+import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail, WorkbenchUserId}
+import org.broadinstitute.dsde.workbench.service.BillingProject.BillingProjectRole
+import org.broadinstitute.dsde.workbench.service.{Google, Orchestration, Rawls, RestException}
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{FreeSpec, Matchers}
+
+import scala.util.Try
+
+
+class BillingApiSpec extends FreeSpec with BillingFixtures with MethodFixtures with SubWorkflowFixtures
+  with Matchers with Eventually {
+
+  /**
+    * This test does
+    *
+    * Given) a registered user who is project owner
+    * and)   the project owner can access to the Google billing account
+    * When)  the project owner is authenticated with access token
+    * Then)  the project owner can create a new Google billing project
+    * and)   the project owner can add another user to Google billing project
+    * and)   the project owner can create a new workspace with Google billing project
+    * and)   the project owner can create new method and method config
+    * and)   the project owner can run an analysis and wait for complete successfully
+    * and)   the project owner can delete method config
+    * and)   the project owner can delete workspace
+    * and)   the project owner can delete Google billing project
+    *
+    */
+  
+  "A user" - {
+    "with a billing account" - {
+      "create a new billing project" in {
+
+        val owner: Credentials = UserPool.chooseProjectOwner
+        implicit val ownerAuthToken: AuthToken = owner.makeAuthToken(AuthTokenScopes.billingScopes)
+
+        // create a new google billing project
+        val billingProjectName = "rawls-billingapispec-" + makeRandomId()
+        register cleanUp Try(deleteBillingProject(billingProjectName)).recover {
+          case _: RestException =>
+        }
+        Rawls.billing.createBillingProject(billingProjectName, ServiceTestConfig.Projects.billingAccountId)
+
+        // verify the google billing project is created and associated with the billing account
+        val associatedBillingAccount = Google.billing.getBillingProjectAccount(billingProjectName)
+        eventually {
+          associatedBillingAccount shouldBe Some(ServiceTestConfig.Projects.billingAccountId)
```

Comment:
this `eventually` block does not seem right... `associatedBillingAccount` is not reevaluated within the block so unless it is right the first time it will never be right.

Reply:
fixed. thank you